operations/generate-ri-report.py

```diff
@@ -0,0 +1,288 @@
+            if "vm_type" in group:
+                bosh_vm_type = group["vm_type"]
+                aws_vm_type = ""
+                for types in cloud_config["vm_types"]:
+                    if types["name"] == bosh_vm_type:
+                        aws_vm_type = types["cloud_properties"]["instance_type"]
+
+                # for some reason this errors out on the first iteration, and I have no idea why.
+                if bosh_vm_type not in vms:
+                    vms[aws_vm_type] = {}
+                    vms[aws_vm_type]["count"] = 0
+                    vms[aws_vm_type]["bosh_vm_type"] = bosh_vm_type
+
+                vms[aws_vm_type]["count"] += group["instances"]
+
+        return vms
+
+
+def args() -> Namespace:
+    args_root = ArgumentParser(usage="Generate a report of deployed VM types for RI purchasing.")
+    args_root.add_argument("--debug", help="Enable debug logging.", action="store_true")
+    args_root.add_argument("--root-path", help="Path where all manifests and cloud-configs reside.",
+                           type=str)
+    args_root.add_argument("--prod-file-pattern", help="Pattern for the production manifests. Defaults to `prodbosh*`.",
+                           type=str,
+                           default="prodbosh-*")
+    args_root.add_argument("--staging-file-pattern",
+                           help="Pattern for the staging manifests. Defaults to `stagingbosh*`.",
+                           type=str,
+                           default="stagingbosh-*")
+    args_root.add_argument("--dev-file-pattern",
+                           help="Pattern for the development manifests. Defaults to `devbosh*`.",
+                           type=str,
+                           default="devbosh-*")
+    args_root.add_argument("--prod-cloud-config-pattern",
+                           help="Pattern for the cloud-config manifest. Defaults to `prodbosh*`.",
+                           type=str,
+                           default="prodbosh-cloud*")
+    args_root.add_argument("--staging-cloud-config-pattern",
+                           help="Pattern for the cloud-config manifests. Defaults to `stagingbosh*`.",
+                           type=str,
+                           default="stagingbosh-cloud*")
+    args_root.add_argument("--dev-cloud-config-pattern",
+                           help="Pattern for the cloud-config manifests. Defaults to `devbosh*`.",
+                           type=str,
+                           default="devbosh-cloud*")
+
+    for g in args_root._action_groups:
+        g._group_actions.sort(key=lambda x: x.dest)
+
+    return args_root.parse_args()
+
+
+def main():
+    Bootstrap(packages=EXTERNAL_DEPENDENCIES)
+    cmd_args = args()
+
+    from logzero import setup_logger
+
+    if cmd_args.debug:
+        logger = setup_logger("main", level=10, formatter=LOG_FORMAT)
+    else:
+        logger = setup_logger("main", level=20, formatter=LOG_FORMAT)
+
+    logger.info("loading files")
```

Comment:
what's wrong with the builtin logging library? I don't think we need to install an additional logging library for a single-file script